File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/SingleQuoteActivity.java
package ro.danserboi.quotesformindandsoul;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;
import android.widget.ToggleButton;

import static ro.danserboi.quotesformindandsoul.MainActivity.recentQuotes;

public class SingleQuoteActivity extends AppCompatActivity {
    private ToggleButton mFavoriteButton;
    private Integer quotePosition = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_single_quote);

        TextView mWordsTextView = findViewById(R.id.wordsSingle);
        TextView mAuthorTextView = findViewById(R.id.authorSingle);
        Button mCopyButton = findViewById(R.id.copyButtonSingle);
        mFavoriteButton = findViewById(R.id.favoriteButtonSingle);
        Button mShareButton = findViewById(R.id.shareButtonSingle);
        Button mChangeFontButton = findViewById(R.id.changeFontButtonSingle);
        ImageButton mChangeBackgroundButton = findViewById(R.id.changeBackgroundButtonSingle);

        Intent detailIntent = getIntent();

        if(detailIntent != null) {
            quotePosition = detailIntent.getIntExtra("position", -1);
            Quote currentQuote = detailIntent.getParcelableExtra("quote");
            assert currentQuote != null;
            String words = currentQuote.getWords();
            String author = currentQuote.getAuthor();
            Boolean isBookmarked = currentQuote.getBookmarked();
            // We add quote to recent list
            recentQuotes.add(currentQuote);
            // We set TextView values and Favorite Button checked state from intent
            mWordsTextView.setText(words);
            mAuthorTextView.setText(author);
            mFavoriteButton.setChecked(isBookmarked);
            // We set OnClickListeners to buttons
            mCopyButton.setOnClickListener(new CopyButtonOnClickListener(this, mWordsTextView, mAuthorTextView));
            mFavoriteButton.setOnClickListener(new BookmarkButtonOnClickListener(this, currentQuote));
            mShareButton.setOnClickListener(new ShareButtonOnClickListener(this, mWordsTextView, mAuthorTextView));
            mChangeFontButton.setOnClickListener(new ChangeFontButtonOnClickListener(this, mWordsTextView, mAuthorTextView));
            mChangeBackgroundButton.setOnClickListener(new ChangeBackgroundOnClickListener(this));
        }
    }

    @Override
    public void onBackPressed() {
        Boolean state = mFavoriteButton.isChecked();
        Intent stateIntent = new Intent();
        stateIntent.putExtra(Utils.EXTRA_STATE, state);
        stateIntent.putExtra(Utils.EXTRA_POSITION, quotePosition);
        setResult(RESULT_OK, stateIntent);
        finish();
        super.onBackPressed();
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/SingleQuoteActivity.kt
package ro.danserboi.quotesformindandsoul

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import android.widget.ImageButton
import android.content.Intent
import android.widget.Button
import android.widget.ToggleButton

class SingleQuoteActivity : AppCompatActivity() {
    private var mFavoriteButton: ToggleButton? = null
    private var quotePosition = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_single_quote)

        val mWordsTextView: TextView = findViewById(R.id.wordsSingle)
        val mAuthorTextView: TextView = findViewById(R.id.authorSingle)
        val mCopyButton: Button = findViewById(R.id.copyButtonSingle)
        mFavoriteButton = findViewById(R.id.favoriteButtonSingle)
        val mShareButton: Button = findViewById(R.id.shareButtonSingle)
        val mChangeFontButton: Button = findViewById(R.id.changeFontButtonSingle)
        val mChangeBackgroundButton: ImageButton = findViewById(R.id.changeBackgroundButtonSingle)

        val detailIntent: Intent = intent
        quotePosition = detailIntent.getIntExtra("position", -1)
        val currentQuote: Quote = detailIntent.getParcelableExtra("quote")!!
        val words: String = currentQuote.words
        val author: String? = currentQuote.author
        val isBookmarked: Boolean = currentQuote.isBookmarked
        // We add quote to recent list
        MainActivity.recentQuotes.add(currentQuote)
        // We set TextView values and Favorite Button checked state from intent
        mWordsTextView.text = words
        mAuthorTextView.text = author
        mFavoriteButton?.isChecked = isBookmarked
        // We set OnClickListeners to buttons
        mCopyButton.setOnClickListener(CopyButtonOnClickListener(this, mWordsTextView, mAuthorTextView))
        mFavoriteButton?.setOnClickListener(BookmarkButtonOnClickListener(this, currentQuote))
        mShareButton.setOnClickListener(ShareButtonOnClickListener(this, mWordsTextView, mAuthorTextView))
        mChangeFontButton.setOnClickListener(ChangeFontButtonOnClickListener(this, mWordsTextView, mAuthorTextView))
        mChangeBackgroundButton.setOnClickListener(ChangeBackgroundOnClickListener(this))
    }

    override fun onBackPressed() {
        val state: Boolean = mFavoriteButton!!.isChecked
        val stateIntent = Intent()
        stateIntent.putExtra(Utils.EXTRA_STATE, state)
        stateIntent.putExtra(Utils.EXTRA_POSITION, quotePosition)
        setResult(RESULT_OK, stateIntent)
        finish()
        super.onBackPressed()
    }
}